practice-one/src/components/Form/index.tsx

```diff
@@ -0,0 +1,257 @@
+            imageUrl: uploadedImageUrl
+        };
+
+        try {
+            const result = await addStudentToAPI(newStudentData);
+            if (result) {
+                onStudentAdd();
+                closeForm();
+            } else {
+                console.error("Failed to add student");
+                setErrorMessage("Failed to add student. Please try again.");
+            }
+        } catch (error) {
+            console.error("Error in onSubmit:", error);
+            setErrorMessage("An error occurred. Please try again.");
+        } finally {
+            setIsSubmitting(false);
+        }
+    };
+
+    const handleImageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files && event.target.files[0];
+        if (file) {
+            const validFormats = ['image/jpeg', 'image/png', 'image/svg+xml'];
+            if (!validFormats.includes(file.type)) {
+                alert("Invalid file format. Only SVG, PNG, and JPG are allowed.");
+                return;
+            }
+
+            setImageFile(file);
+            setImagePreviewUrl(URL.createObjectURL(file));
+        }
+    };
+
+    return (
+        <form onSubmit={handleSubmit(onSubmit)} className="add-student-form">
+            <div className="form-header">
+                <h2 className="heading-form">Add Student</h2>
+                <div className="link-group">
+                    <a href="#" className="link-group-item">Manually</a>
+                    <a href="#" className="link-group-item">Import CSV</a>
+                </div>
+            </div>
+            <div className="form-body">
+                <div className="form-body-item">
+                    <div className="student-name item">
+                        <p className="item-title">Name</p>
+                        <Controller
+                            name="name"
+                            control={control}
+                            rules={validationRules.name}
+                            render={({ field }) => (
+                                <Input
+                                    {...field}
+                                    className="default"
+                                    type="text"
+                                    error={errors.name ? errors.name.message : null}
+                                />
+                            )}
+                        />
+                    </div>
+                    <div className="student-class item">
+                        <DropdownSelect
+                            name="classes"
+                            label="Class"
+                            options={CLASS_OPTIONS}
+                            control={control}
+                            requiredMessage="Please choose class"
+                        />
+                        {errors.classes && <span className="error-message">{errors.classes.message}</span>}
```

Comment:
why the DropdownSelect have error param why need to create error in here
please check for all